lib/src/core/routing/routing.dart

```diff
@@ -8,22 +13,61 @@ enum AppRoutes {
   course,
 }
 
-routerDelegate({String? initialPath}) => BeamerDelegate(
-      initialPath: initialPath ?? '/',
-      //TODO -LOW/MEDIUM- remove animation just on web
-      transitionDelegate: const NoAnimationTransitionDelegate(),
-      locationBuilder: RoutesLocationBuilder(
-        routes: {
-          '/': (_, __, ___) => const HomePage(),
-          '/archive': (_, __, ___) => const ArchivePage(),
-          '/course/:path': (_, state, __) {
-            final path = state.pathParameters['path']!;
-            return CoursePage(path);
-          },
-          '/article/:path': ((_, state, __) {
-            final path = (state.pathParameters['path']!);
-            return ArticlePage(path: path);
-          }),
-        },
-      ),
-    );
+final routerDelegate = BeamerDelegate(
+  locationBuilder: RoutesLocationBuilder(
+    routes: {
+      // Return either Widgets or BeamPages if more customization is needed
+      '/:locale': (_, state, ___) {
+        final languageCode = (state.pathParameters['locale']!);
+        return LocaleWrapper(
+          languageCode: languageCode,
+          child: const HomePage(),
+        );
+      },
+      '/:locale/archive': (_, state, ___) {
+        final languageCode = (state.pathParameters['locale']!);
+        return LocaleWrapper(
+          languageCode: languageCode,
+          child: const ArchivePage(),
+        );
+      },
+      '/:locale/course/:path': (_, state, __) {
+        final path = state.pathParameters['path']!;
+        final languageCode = (state.pathParameters['locale']!);
+        return LocaleWrapper(
+          languageCode: languageCode,
+          child: CoursePage(path),
+        );
+      },
+      '/:locale/article/:path': (_, state, __) {
+        final path = state.pathParameters['path']!;
+        final languageCode = (state.pathParameters['locale']!);
+        return LocaleWrapper(
+          languageCode: languageCode,
+          child: ArticlePage(path: path),
+        );
+      },
+    },
+  ),
+  initialPath: '/en',
+);
+
+class LocaleWrapper extends StatelessWidget {
+  const LocaleWrapper({
+    super.key,
+    required this.languageCode,
+    required this.child,
+  });
+  final String languageCode;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final locale = Locale(languageCode);
+    if (AppLocalizations.supportedLocales.contains(locale)) {
+      context.read<LocaleCubit>().setLocale(Locale(languageCode));
+      return child;
+    }
+    return const NotFoundScreen();
```

Comment:
As said before, it the language isn't supported will return a NotFoundScreen

Reply:
Cooooool!